include/dlaf/multiplication/general.h

```diff
@@ -66,4 +67,32 @@ void generalSubMatrix(const SizeType a, const SizeType b, const blas::Op opA, co
   else
     DLAF_UNIMPLEMENTED(opA, opB);
 }
+
+template <Backend B, Device D, class T>
+void generalSubMatrixK(comm::CommunicatorGrid grid, const SizeType a, const SizeType b, const SizeType k,
```

Comment:
```suggestion
void generalSubMatrixSquare(comm::CommunicatorGrid grid, const SizeType a, const SizeType b, const SizeType k,
```
, at least if I interpret the `K` as referring to the inputs being `k`×`k` matrices?

Reply:
Just to give a bit more context, but you are already on the good track.

In the best case, this algorithm will accept three main parameters:
- idx_begin
- idx_last
- k

Given those, it will compute the gemm between the square matrices using A, B and C but just in the range of tiles `[idx_begin, idx_last]`, so it will be by construction a gemm over square matrices.

The `k` parameter, which we are not still sure it will be implemented in this first version, opens to an optimisation where just the first k rows/cols in these square sub-matrices are actually used for the gemm.

Reply:
After @albestro's answer and speaking to him offline I now understand why `generalSubMatrixSquare` doesn't make sense as a name, or at least doesn't make sense only for the `K` version: both `generalSubMatrix` and `generalSubMatrixK` operate on square submatrices by construction. An important distinction that I also didn't understand first is that `a` and `b` are tile indices, while `k` is an element index (or size). So I retract at least my original renaming suggestion. Perhaps some other name would still be more descriptive but I'm also ok with leaving the name as it is right now.